resources/shaders/lighting.hlsl

```diff
@@ -84,7 +92,7 @@ float3 shade_pixel(float3 pos, float3 V, float3 albedo, float metallic, float ro
 	return ambient + res + emissive;
 }
 
-float3 shade_light(float3 pos, float3 V, float3 albedo, float3 normal, float metallic, float roughness, Light light, inout uint rand_seed, uint depth)
+float3 shade_light(float3 pos, float3 V, float3 albedo, float3 normal, float metallic, float roughness, Light light, inout uint rand_seed, uint depth, uint calling_pass)
```

Comment:
this is now inconsistent previously you passes 2 ints now a enum type thing. Not very nice imo.

Reply:
Enum used everywhere now, and also renamed.